### DISCUSSION TITLE
Run / Build

### DISCUSSION THREAD

**Discussion by ghost** [Sketchware-Pro/Sketchware-Pro#128] [Ideas]
**Hey There**,
Can you add a option to run project to just check java errors, Where as Build Completely builds the apk

Well, you can suggest to cancel build after jave compiled but in some huge projects Cancelling Build Is A Pain.

Thank You

**Comment by j-px** [comment]
hmm, usually dexing or ProGuarding/etc. takes the most time. Compiling Java classes needs R.java generated by AAPT2/AAPT2, so we can't skip that part. (caching would be possible, but may be confusing to some)

**Reply by ghost** [reply]
Mm... Stop Build After java compiled 

// Set Option in build Settings
      If run is selected run =true;
      If build is selected  run =false;

//after java compiled 
If(run){
//stop build,don't move to next step}
else{
//continue build};

Will it work bro

**Reply by j-px** [reply]
well that _would_ work, but it's not really useful. If your project has any errors, it'd stop compiling right "after" ecj. you can let it compile, the "cancel" the process when the dialog says "Java is compiling..." and if there are errors, they'd pop up like a regular error

**Reply by ghost** [reply]
Noice Idea
